src/Application/App.hpp

```diff
@@ -327,6 +331,11 @@ class App : public KPController, public TaskObserver {
         }
 
         for (auto v : task.valves) {
+            if(v > vm.valves.size()){
```

Comment:
This is a defensive check from the sampler firmware, consider adding a similar check to the UI.